src/GraphBLAS-sharp/Methods/Scalar.fs

```diff
@@ -0,0 +1,39 @@
+namespace GraphBLAS.FSharp
+
+open Brahma.FSharp.OpenCL.WorkflowBuilder.Basic
+
+[<CompilationRepresentation(CompilationRepresentationFlags.ModuleSuffix)>]
+module Scalar =
+
+    (*
+        constructors
+    *)
+
+    let internal createFromArray (value: 'a[]) : GraphblasEvaluation<Scalar<'a>> = graphblas { return Scalar.FromArray(value) }
+
+    let create (value: 'a) : GraphblasEvaluation<Scalar<'a>> = createFromArray [|value|]
+
+    (*
+        methods
+    *)
+
+    let copy (scalar: Scalar<'a>) : GraphblasEvaluation<Scalar<'a>> = failwith "Not Implemented yet"
+
+    let synchronize (scalar: Scalar<'a>) : GraphblasEvaluation<unit> =
+        opencl {
+            let! _ = ToHost scalar.Value
+
+            return ()
+        }
+        |> EvalGB.fromCl
+
+    (*
+        assignment and extraction
+    *)
+
+    let extractValue (scalar: Scalar<'a>) : GraphblasEvaluation<'a> = graphblas { return scalar.Value.[0] }
```

Comment:
This will only work properly if you sync the scalar before the operation. So it makes sense to add `exportValue` operation (synchronize + return underlying value) rather than `extractValue`.